gcloud-sdk/src/token_source/ext_creds_source/mod.rs

```diff
@@ -139,3 +184,175 @@ fn subject_token_from_json(
         .into()
     })
 }
+
+#[cfg(feature = "external-account-aws")]
+mod aws {
+    use crate::error::Error;
+    use crate::error::ErrorKind;
+    use aws_config::Region;
+    use aws_credential_types::provider::ProvideCredentials;
+    use aws_credential_types::Credentials;
+    use aws_sigv4::http_request::{
+        SignableBody, SignatureLocation, SigningParams, SigningSettings,
+    };
+    use aws_sigv4::sign::v4::SigningParams as V4SigningParams;
+    use hyper::http::{Method, Request};
+    use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
+    use secret_vault_value::SecretValue;
+    use serde::Serialize;
+    use std::time::SystemTime;
+
+    pub async fn subject_token_aws(
+        regional_cred_verification_url: &str,
+        credentials: Credentials,
+        region: Region,
+        sign_at: SystemTime,
+        audience: &str,
+    ) -> crate::error::Result<SecretValue> {
+        let identity = credentials.into();
+        let signature_time = sign_at;
+
+        let mut signing_settings = SigningSettings::default();
+        signing_settings.signature_location = SignatureLocation::Headers;
+        let v4_signing_params = V4SigningParams::builder()
+            .name("sts")
+            .identity(&identity)
+            .region(region.as_ref())
+            .time(signature_time)
+            .settings(signing_settings)
+            .build()
+            .map_err(|e| Error::from(ErrorKind::ExternalCredsSourceError(e.to_string())))?;
+        let params = SigningParams::V4(v4_signing_params);
+
+        let regional_cred_verification_url =
+            regional_cred_verification_url.replace("{region}", region.as_ref());
+        let subject_token_url = regional_cred_verification_url;
+        let url = url::Url::parse(&subject_token_url).unwrap();
```

Comment:
Would be great to handle errors instead of `unwrap()` here.

Reply:
Thank you for pointing this out.
Fixed in https://github.com/abdolence/gcloud-sdk-rs/pull/172/commits/fa3979a8d1a376dd85968330d48e7580eb547d81